plugins/modules/ipa_getkeytab.py

```diff
@@ -0,0 +1,253 @@
+  force:
+    description:
+      - Force recreation if exists already.
+    type: bool
+requirements:
+  - freeipa-client
+  - Managed host is FreeIPA client
+extends_documentation_fragment:
+  - community.general.attributes
+'''
+
+EXAMPLES = r'''
+- name: Get kerberos ticket
+  ansible.builtin.shell: kinit admin
+  args:
+    stdin: "{{ aldpro_admin_password }}"
+  changed_when: true
+
+- name: Create keytab
+  community.general.ipa_getkeytab:
+    path: /etc/ipa/test.keytab
+    principal: HTTP/freeipa-dc02.ipa.test
+    ipa_server: freeipa-dc01.ipa.test
+
+- name: Retrieve already existing keytab
+  community.general.ipa_getkeytab:
+    path: /etc/ipa/test.keytab
+    principal: HTTP/freeipa-dc02.ipa.test
+    ipa_server: freeipa-dc01.ipa.test
+    retrieve_mode: true
+
+- name: Force keytab recreation
+  community.general.ipa_getkeytab:
+    path: /etc/ipa/test.keytab
+    principal: HTTP/freeipa-dc02.ipa.test
+    ipa_server: freeipa-dc01.ipa.test
+    force: true
+'''
+
+import os
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.community.general.plugins.module_utils.cmd_runner import CmdRunner, cmd_runner_fmt
+
+
+class IPAKeytab(object):
+    def __init__(self, module, **kwargs):
+        self.module = module
+        self.path = kwargs['path']
+        self.state = kwargs['state']
+        self.principal = kwargs['principal']
+        self.ipa_server = kwargs['ipa_server']
+        self.ldap_uri = kwargs['ldap_uri']
+        self.bind_dn = kwargs['bind_dn']
+        self.bind_pw = kwargs['bind_pw']
+        self.password = kwargs['password']
+        self.ca_certificate = kwargs['ca_certificate']
+        self.sasl_mech = kwargs['sasl_mech']
+        self.retrieve_mode = kwargs['retrieve_mode']
+        self.encryption_types = kwargs['encryption_types']
+
+        self.executable = [module.get_bin_path('ipa-getkeytab', True)]
+        self.runner = CmdRunner(
+            module,
+            command=self.executable,
+            arg_formats=dict(
+                retrieve_mode=cmd_runner_fmt.as_bool('--retrieve'),
+                path=cmd_runner_fmt.as_opt_val('--keytab'),
+                ipa_server=cmd_runner_fmt.as_opt_val('--server'),
+                principal=cmd_runner_fmt.as_opt_val('--principal'),
+                ldap_uri=cmd_runner_fmt.as_opt_val('--ldapuri'),
+                bind_dn=cmd_runner_fmt.as_opt_val('--binddn'),
+                bind_pw=cmd_runner_fmt.as_opt_val('--bindpw'),
+                password=cmd_runner_fmt.as_opt_val('--password'),
+                ca_certificate=cmd_runner_fmt.as_opt_val('--cacert'),
+                sasl_mech=cmd_runner_fmt.as_opt_val('--mech'),
+                encryption_types=cmd_runner_fmt.as_opt_val('--enctypes'),
+            )
+        )
+
+    def _exec(self, run_in_check_mode=False, check_rc=True):
+        if not self.module.check_mode or (self.module.check_mode and run_in_check_mode):
+            params = dict(self.module.params)
```

Comment:
I am not sure these parameters make sense. They are never really used and module's code is not reused. Code is simpler without them.
Additionally, you might want to take a look at StateModuleHelper, see https://docs.ansible.com/ansible/latest/collections/community/general/docsite/guide_modulehelper.html for details on how to use it. There is a number of modules in this collection using that you can look up to for examples of actual usage.